components/script/dom/node.rs

```diff
@@ -768,7 +768,11 @@ impl Node {
     }
 
     pub fn is_in_html_doc(&self) -> bool {
-        self.owner_doc().is_html_document()
+        self.is_in_doc() && self.owner_doc().is_html_document()
```

Comment:
Unfortunately this needs to return true if `is_in_doc()` is false. Maybe you should just inline it into its only caller, to avoid confusion.